app/models/tslr_claim.rb

```diff
@@ -134,4 +143,26 @@ def ni_number_is_correct_format
     errors.add(:national_insurance_number, "Enter a National Insurance number in the correct format") \
       if national_insurance_number.present? && !normalised_ni_number.match(/\A[a-z]{2}[0-9]{6}[a-d]{1}\Z/i)
   end
+
+  def normalise_account_number
+    self.account_number = normalised_bank_detail(account_number)
+  end
+
+  def normalise_account_sort_code
+    self.account_sort_code = normalised_bank_detail(account_sort_code)
```

Comment:
Don't think it needs to change, but I think the methods you were looking for are `account_number_changed?` and `account_sort_code_changed?`, same as you used in the callback declaration.

Reply:
Yeah, I had those. I was thinking I might be able to do something like:

`before_save :normalise_bank_details, if: :bank_sort_code_changed? || :bank_account_number_changed?`

But looking at it now I'm not sure it's as clear.